aws/resource_aws_s3control_bucket_lifecycle_configuration_test.go

```diff
@@ -0,0 +1,750 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Date(rName, date1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketLifecycleConfigurationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"expiration.#":      "1",
+						"expiration.0.date": date1,
+					}),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Date(rName, date2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketLifecycleConfigurationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"expiration.#":      "1",
+						"expiration.0.date": date2,
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSS3ControlBucketLifecycleConfiguration_Rule_Expiration_Days(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_s3control_bucket_lifecycle_configuration.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSOutpostsOutposts(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSS3ControlBucketLifecycleConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Days(rName, 7),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketLifecycleConfigurationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"expiration.#":      "1",
+						"expiration.0.days": "7",
+					}),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSS3ControlBucketLifecycleConfigurationConfig_Rule_Expiration_Days(rName, 30),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketLifecycleConfigurationExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"expiration.#":      "1",
+						"expiration.0.days": "30",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSS3ControlBucketLifecycleConfiguration_Rule_Expiration_ExpiredObjectDeleteMarker(t *testing.T) {
+	TestAccSkip(t, "S3 on Outposts does not error or save it in the API when receiving this parameter")
```

Comment:
given this behavior - should this be documented in the resource docs for users?

Reply:
Hopefully my out of band answer sufficed here, but we can certainly add this post merge if necessary. 👍